### PR TITLE
Feature/fields

### PR DESCRIPTION
**Pull request by blackshadev**
Added a way to add fields to the query

docs: 
- https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-fields.html
- https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html

### CODE REVIEW

File: src/Domain/IndexManagement/IndexConfiguration.php [hunk 1]
```diff
@@ -17,11 +17,6 @@ private function __construct(string $name)
         $this->name = $name;
     }
 
-    public static function empty(string $name): self
```
Comment (Jeroen-G) [inline src/Domain/IndexManagement/IndexConfiguration.php:20]: Why did this go?

**Comment by Jeroen-G** [comment]
Do you have a link to the right ES docs for these fields? I'd like to write some documentation on it.

File: src/Domain/IndexManagement/IndexConfiguration.php (hunk 1 above)
Comment (blackshadev) [reply src/Domain/IndexManagement/IndexConfiguration.php:20]: Unused. mutation testing detected it was unused (and untested)

**Comment by blackshadev** [comment]
> Do you have a link to the right ES docs for these fields? I'd like to write some documentation on it.

Updated description with the docs. most relevant is:
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html

**Pull request merged**